### PR TITLE
Adding Local Setup and Promtail configuration

### PR DESCRIPTION
**Pull request by lieberlois**

### CODE REVIEW

File: stack/kubernetes/tools/local/loki.yaml
[the remark this answers is not in the document]
Comment (lieberlois) [reply stack/kubernetes/tools/local/loki.yaml:1]: Could you elaborate on that? :)

File: stack/kubernetes/tools/local/promtail.yaml
[the remark this answers is not in the document]
Comment (lieberlois) [reply stack/kubernetes/tools/local/promtail.yaml:1]: See other comment

File: stack/kubernetes/tools/local/tempo.yaml
[the remark this answers is not in the document]
Comment (lieberlois) [reply stack/kubernetes/tools/local/tempo.yaml:1]: See other comment

File: stack/kubernetes/tools/local/grafana.yaml
[the remark this answers is not in the document]
Comment (lieberlois) [reply stack/kubernetes/tools/local/grafana.yaml:1]: See other comment

File: stack/kubernetes/tools/local/loki.yaml
[the remark this answers is not in the document]
Comment (ukinimod) [reply stack/kubernetes/tools/local/loki.yaml:1]: the local helm config for the application stack is in /local/application. I would have expected the local helm config for monitoring in /local/tools/monitoring as they are in /stack/tools/monitoring in the gke case. I think to help orientate in the repo the structure of the files should be basically the same for the case of remote and local deployment.

File: local/kubernetes/tools/logging/kibana-index/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM appropriate/curl
```
Comment (BlacCello) [inline local/kubernetes/tools/logging/kibana-index/Dockerfile:1]: Is it required to duplicate this Dockerfile and all the scripts for local setup?

File: local/kubernetes/infrastructure/namespaces.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+apiVersion: v1
```
Comment (BlacCello) [inline local/kubernetes/infrastructure/namespaces.yaml:1]: In the GKE setup, the namespaces are created via helm --create-namespace. Do you think it makes sense to align this?

**Review by BlacCello — COMMENTED** [review]
I am fine for now with the basic setup (not tested yet), but we have to merge a lot of configuration files and only change some settings via helm value files (this might lead to making shell scripts configurable with arguments).

**Comment by BlacCello** [comment]
Can you please add a README for the local setup? This makes testing this PR easier.

File: stack/kubernetes/tools/local/loki.yaml
[the remark this answers is not in the document]
Comment (BlacCello) [reply stack/kubernetes/tools/local/loki.yaml:1]: The structure was aligned.

File: local/30_build.sh [hunk 3]
```diff
@@ -2,6 +2,8 @@
 set -euo pipefail
 
 export GKE_REGISTRY=local
-export GITHUB_SHA="$(git rev-parse --short HEAD)-snapshot"
+
+GITHUB_SHA="$(git rev-parse --short HEAD)-snapshot"
```
[the remark this answers is not in the document]
Comment (BlacCello) [reply local/30_build.sh]: Should be handled in https://github.com/feature-creeps/observability-workshop/issues/135

File: local/kubernetes/infrastructure/namespaces.yaml (hunk 2 above)
Comment (BlacCello) [reply local/kubernetes/infrastructure/namespaces.yaml:1]: See https://github.com/feature-creeps/observability-workshop/issues/137

File: local/kubernetes/tools/logging/kibana-index/Dockerfile (hunk 1 above)
Comment (BlacCello) [reply local/kubernetes/tools/logging/kibana-index/Dockerfile:1]: See #138

**Pull request merged**